browser/android/brave_sync_worker.cc

```diff
@@ -233,14 +233,6 @@ void BraveSyncWorker::SetEncryptionPassphrase(syncer::SyncService* service) {
 
   syncer::SyncUserSettings* sync_user_settings = service->GetUserSettings();
   DCHECK(!sync_user_settings->IsPassphraseRequired());
-
-  if (sync_user_settings->IsCustomPassphraseAllowed() &&
-      !sync_user_settings->IsUsingExplicitPassphrase() &&
-      !sync_user_settings->IsTrustedVaultKeyRequired()) {
-    sync_user_settings->SetEncryptionPassphrase(this->passphrase_);
```

Comment:
We must remove `ProfileMetrics::LogProfileSyncInfo()` but we can't touch `SetEncryptionPassphrase`. Sync must not work as it is now, because it will not be able to decrypt data from the server. 

I am building branch and will create PR with a fix

Reply:
PR is created https://github.com/brave/brave-core/pull/19333 
iOS/desktop are not affected.

The strange thing that even without this fix sync on cr116 works. Obviously the wrong encryption key is used, keystore vs custom:
before|after
---|---
 ![image](https://github.com/brave/brave-core/assets/24739341/4025eba3-0177-4e6f-a113-abd55bdd3dcd) | ![image](https://github.com/brave/brave-core/assets/24739341/9fface52-0827-4933-a412-fc47f37857ef)